frontend/public/components/utils/webhooks.tsx

```diff
@@ -14,58 +20,114 @@ enum TriggerTypes {
   ImageChange = 'ImageChange',
 }
 const webhookTriggers = new Set<TriggerTypes>([TriggerTypes.Bitbucket, TriggerTypes.Generic, TriggerTypes.GitHub, TriggerTypes.GitLab]);
-const getTriggerProperty = trigger => trigger.type.toLowerCase();
+const getTriggerProperty = (trigger: WebhookTrigger) => trigger.type.toLowerCase();
 
-export const WebhookTriggers: React.SFC<WebhookTriggersProps> = ({ resource }) => {
+const getTableColumnClasses = (canGetSecret: boolean) => {
+  if (canGetSecret) {
+    return [
+      classNames('col-lg-2', 'col-md-4', 'col-sm-4', 'col-xs-6'),
+      classNames('col-lg-6', 'hidden-md', 'hidden-sm', 'hidden-xs', 'co-break-all'),
+      classNames('col-lg-2', 'col-md-4 ', 'col-sm-4', 'hidden-xs'),
+      classNames('col-lg-2', 'col-md-4', 'col-sm-4', 'col-xs-6'),
+    ];
+  }
+  return [
+    classNames('col-sm-2'),
+    classNames('col-sm-7', 'co-break-all'),
+    classNames('col-sm-3', 'hidden-xs'),
+    classNames('hidden'),
+  ];
+};
+
+export const WebhookTriggers: React.FC<WebhookTriggersProps> = props => {
+  const { resource } = props;
   const { name, namespace } = resource.metadata;
   const { triggers } = resource.spec;
+  const canGetSecret = useAccessReview({
+    group: SecretModel.apiGroup,
+    resource: SecretModel.plural,
+    verb: 'get',
+    namespace,
+  });
+  const tableColumnClasses = getTableColumnClasses(canGetSecret);
   const webhooks = _.filter(triggers, trigger => webhookTriggers.has(trigger.type));
   if (_.isEmpty(webhooks)) {
     return null;
   }
 
-  const getWebhookURL = trigger => {
+  const getWebhookURL = (trigger: WebhookTrigger, secret?: string) => {
     const triggerProperty = getTriggerProperty(trigger);
-
-    // FIXME: Consider showing the secret in the table so that users can copy the real URL. Maybe a show/hide toggle like secrets?
-    // const secret = _.get(trigger, [triggerProperty, 'secret'], '<secret>');
-    // return `${kubeAPIServerURL}/apis/build.openshift.io/v1/namespaces/${namespace}/buildconfigs/${name}/webhooks/${secret}/${triggerProperty}`;
-
-    return `${kubeAPIServerURL}/apis/build.openshift.io/v1/namespaces/${namespace}/buildconfigs/${name}/webhooks/<secret>/${triggerProperty}`;
+    return `${kubeAPIServerURL}/apis/build.openshift.io/v1/namespaces/${namespace}/buildconfigs/${name}/webhooks/${secret ? secret : '<secret>'}/${triggerProperty}`;
   };
 
-  const getSecretReference = trigger => {
+  const getSecretReference = (trigger: WebhookTrigger) => {
     const triggerProperty = getTriggerProperty(trigger);
+    if (_.has(trigger, [triggerProperty, 'secret'])) {
```

Comment:
I don’t think this change is necessary since we handle the no secret reference case below